frontend/public/components/build.tsx

```diff
@@ -61,13 +61,42 @@ export const BuildsDetails: React.SFC<BuildsDetailsProps> = ({obj: build}) => {
   </div>;
 };
 
-const envPath = ['spec', 'strategy', 'sourceStrategy'];
-const environmentComponent = (props) => <EnvironmentPage
-  obj={props.obj}
-  rawEnvData={props.obj.spec.strategy.sourceStrategy}
-  envPath={envPath}
-  readOnly={true}
-/>;
+export const getStrategyType = (strategy) => {
+  switch (strategy.type) {
+    case 'Docker':
+      return 'dockerStrategy';
+    case 'Custom':
+      return 'customStrategy';
+    case 'JenkinsPipeline':
+      return 'jenkinsPipelineStrategy';
+    case 'Source':
+      return 'sourceStrategy';
+    default:
+      return null;
+  }
+};
+
+export const getEnvPath = (props) => {
+  const strategyType = getStrategyType(props.obj.spec.strategy);
+  return strategyType ? ['spec', 'strategy', strategyType] : null;
+};
+
+const environmentComponent = (props) => {
```

Comment:
If you export this as `buildEnvironmentComponent`, you can use it directly in build-config.tsx. Then you don't repeat the code or need to export the other two helper functions.

Reply:
They are two slightly different components. The build components are read/only and the build config is read/write

Reply:
Ah, sorry, I see. If that's the only difference, you might add

```js
// Environment variables are only editable on build configs, not builds.
const readOnly = obj.kind === 'Build';
```